app/src/main/java/nodomain/freeyourgadget/gadgetbridge/util/RtlUtils.java

```diff
@@ -0,0 +1,466 @@
+package nodomain.freeyourgadget.gadgetbridge.util;
+
+import android.util.Log;
+import android.util.Pair;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+import nodomain.freeyourgadget.gadgetbridge.GBApplication;
+
+class RtlUtils {
+    /**
+     * Checks the status of right-to-left option
+     * @return true if right-to-left option is On, and false, if Off or not exist
+     */
+    public static boolean contextualSupport()
+    {
+        return GBApplication.getPrefs().getBoolean("contextualArabic", false);
+    }
+
+    //map with brackets chars to change there direction
+    private static Map<Character, Character> directionSignsMap = new HashMap<Character, Character>(){
+        {
+            put('(', ')'); put(')', '('); put('[', ']'); put(']', '['); put('{','}'); put('}','{');
+
+
+        }
+    };
+
+    //list of unicode ranges of rtl chars
+    private static ArrayList<Pair<Character, Character>> hebrewRange = new ArrayList<Pair<Character, Character>>() {
+        {
+            add(new Pair<Character, Character>('\u0590', '\u05F4'));
+            add(new Pair<Character, Character>('\uFB1D', '\uFB4F'));
+        }
+    };
+
+    //list of unicode ranges of rtl chars
+    private static ArrayList<Pair<Character, Character>> arabicRange = new ArrayList<Pair<Character, Character>>() {
+        {
+            add(new Pair<Character, Character>('\u0600', '\u06FF'));
+            add(new Pair<Character, Character>('\u0750', '\u077F'));
+            add(new Pair<Character, Character>('\u08A0', '\u08FF'));
+            add(new Pair<Character, Character>('\uFB50', '\uFDFF'));
+            add(new Pair<Character, Character>('\uFE70', '\uFEFF'));
+        }
+    };
+
+    //list of unicode ranges of rtl chars
+    private static ArrayList<Pair<Character, Character>> rtlRange = new ArrayList<Pair<Character, Character>>() {
+        {
+            addAll(hebrewRange);
+            addAll(arabicRange);
+        }
+    };
+
+    /**
+     * @return true if the char is in the rtl range, otherwise false
+     */
+    static Boolean isHebrew(char c){
+        for (Pair<Character, Character> rang: hebrewRange) {
+            if (rang.first <= c && c <= rang.second) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * @return true if the char is in the rtl range, otherwise false
+     */
+    static Boolean isArabic(char c){
+        for (Pair<Character, Character> rang: arabicRange) {
+            if (rang.first <= c && c <= rang.second) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * @return true if the char is in the rtl range, otherwise false
+     */
+    static Boolean isRtl(char c){
+        for (Pair<Character, Character> rang: rtlRange) {
```

Comment:
Could we use `Character.getDirectionality()` instead?

Reply:
I will try.
From what I see, I'm not sure it's cover *every* Hebrew and Arabic characters, but I'm not sure if those characters are really important.
I will do some testing.

Reply:
Thanks! You certainly know more about this topic than me. I only learned about this method yesterday while reading up on this.

Reply:
I investigate this last night, and it seems I can use it.
What I changed is the implementation of isRtl/isHebrew and such to use this function instead of searching.

Reply:
Done